extensions/libarchive/MergeContent.cpp

```diff
@@ -342,6 +352,50 @@ std::shared_ptr<core::FlowFile> ZipMerge::merge(core::ProcessContext *context, c
   return flowFile;
 }
 
+void AttributeMerger::mergeAttributes(core::ProcessSession *session, std::shared_ptr<core::FlowFile> &merge_flow) {
+  for (const auto& pair : getMergedAttributes()) {
+    session->putAttribute(merge_flow, pair.first, pair.second);
+  }
+}
+
+std::map<std::string, std::string> AttributeMerger::getMergedAttributes() {
+  std::map<std::string, std::string> merged_attributes;
```

Comment:
Maybe `std::accumulate` could make this simpler:
```c++
if(flows_.empty()) {
  return {}
}
auto merge_attributes = [&] (
    const std::map<std::string, std::string>& merged_attributes,
    const std::shared_ptr<core::FlowFile>& flow)
    { processFlowFile(flow, merged_attributes); }
return std::accumulate(std::next(flows_.cbegin()), flows.cend(), flows_.begin()->getAttributes(), merge_attributes);

Reply:
I tried this solution before as @szaszm also suggested it, and it was working and was a quite elegant solution, but I dropped it as the [cppreference](https://en.cppreference.com/w/cpp/algorithm/accumulate) suggested that std::move is only used in its implementation from C++20, so copying every temporal map result when there is a large number of flow files may be slow.

Reply:
You mean this move?
![Screenshot 2020-08-13 at 18 23 20](https://user-images.githubusercontent.com/64011968/90160450-11b10900-dd92-11ea-8515-eaf0ca020af9.png)
The lambda captures by reference, so it should not be a problem.

The assembly generated is also the same, see here:
https://quick-bench.com/q/J7uVr8A_Ib62WQrJm9QXm7zK5Wo

Reply:
There was an error in the code you sent, both versions were called with manualLooping() :) But there was no difference in the result after fixing it.

I was thinking about that move, but actually I was thinking referring to the other case, the Keep All Unique Attributes scenario. In that case we currently merge the attributes in place, but using the std::accumulate we cannot do that, and as there is no move support, I cannot consume the parameter, so I have to make a copy of that map. You can check the difference here:
https://quick-bench.com/q/RauiICzYXNNBplQmLFj2bZ7z14E

Btw thanks for this tool, I didn't know about this and it's pretty handy :)

Reply:
I've got this down to near-equal by introducing a pointer-indirection to the map and processing it in place anyway, but this probably defeats the purpose. (Let me join the quick-bench party! :smile: )
https://quick-bench.com/q/7kxa6NSSuo9Dki-J0Mllh3vABeQ

Reply:
A bit circumvented solution, but I am okay with it. At this point I have no preference one way or another so I can add it in the next commit if there are no objections.

Reply:
Please don't. This was just a demonstration, but it defeats the purpose of accumulate by circumventing the assignment part. If you prefer STL algorithms, a for_each should be able to do the same.